apps/typeform/frontend/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,239 @@
+import get from 'lodash/get';
+import {
+  Heading,
+  Paragraph,
+  Typography,
+  TextLink,
+  Select,
+  Option,
+  FormLabel
+} from '@contentful/forma-36-react-components';
+import FieldSelector from './FieldSelector';
+import {
+  CompatibleFields,
+  ContentType,
+  Hash,
+  EditorInterface,
+  InstallationParameters,
+  SelectedFields,
+  WorkspaceOption,
+  WorkspacesResponse
+} from '../typings';
+import {
+  getCompatibleFields,
+  editorInterfacesToSelectedFields,
+  selectedFieldsToTargetState,
+  validateParameters
+} from '../utils';
+import { styles } from './styles';
+
+// @ts-ignore 2307
+import logo from './config-screen-logo.svg';
+
+interface Props {
+  sdk: AppExtensionSDK;
+  expireSoon: boolean;
+}
+
+interface State {
+  selectedWorkspaceId: string;
+  accessToken: string;
+  workspaces: WorkspaceOption[];
+  contentTypes: ContentType[];
+  selectedContentTypes: string[];
+  selectedFields: SelectedFields;
+  compatibleFields: CompatibleFields;
+}
+
+export class AppConfig extends React.Component<Props, State> {
+  state: State = {
+    contentTypes: [],
+    compatibleFields: {},
+    workspaces: [],
+    selectedContentTypes: [],
+    selectedFields: {},
+    selectedWorkspaceId: '',
+    accessToken: (window.localStorage.getItem('token') as string) || ''
+  };
+
+  async componentDidMount() {
+    const { sdk } = this.props;
+
+    sdk.app.onConfigure(this.onAppConfigure);
+
+    const [contentTypesResponse, eisResponse, paramsResponse] = await Promise.all([
+      sdk.space.getContentTypes(),
+      sdk.space.getEditorInterfaces(),
+      sdk.app.getParameters(),
+      this.fetchWorkspaces()
+    ]);
+
+    const contentTypes = (contentTypesResponse as Hash).items as ContentType[];
+    const editorInterfaces = (eisResponse as Hash).items as EditorInterface[];
+    const compatibleFields = getCompatibleFields(contentTypes);
+    const filteredContentTypes = contentTypes.filter(ct => {
+      const fields = compatibleFields[ct.sys.id];
+      return fields && fields.length > 0;
+    });
+
+    const parameters: InstallationParameters = paramsResponse as InstallationParameters;
+
+    this.setState(
+      {
+        selectedWorkspaceId: get(parameters, ['selectedWorkspaceId'], ''),
+        compatibleFields,
+        contentTypes: filteredContentTypes,
+        selectedFields: editorInterfacesToSelectedFields(editorInterfaces, sdk.ids.app)
+      },
+      () => sdk.app.setReady()
+    );
+  }
+
+  fetchWorkspaces = async () => {
+    try {
+      const response = await fetch(`/workspaces`, {
+        headers: {
+          Authorization: `Bearer ${this.state.accessToken}`
+        }
+      });
+      const result: WorkspacesResponse = await response.json();
+      this.setState({ workspaces: this.normalizeWorkspaceResponse(result) });
+    } catch (_error) {
+      this.props.sdk.notifier.error(
+        'There was a problem while fetching your Typeform workspaces. Please try again.'
```

Comment:
Minor grammar correction:
```suggestion
        'There was a problem fetching your Typeform workspaces. Please try again.'
```